xknxproject/util.py

```diff
@@ -7,10 +11,27 @@ def parse_dpt_types(dpt_types: list[str]) -> dict[str, int]:
     if len(dpt_types) == 0:
         return {}
 
-    dpt_type: str = dpt_types[len(dpt_types) - 1]
+    dpt_type: str = dpt_types[-1]
     if MAIN_DPT in dpt_type:
         return {"main": int(dpt_type.split("-")[1])}
     if MAIN_AND_SUB_DPT in dpt_type:
         return {"main": int(dpt_type.split("-")[1]), "sub": int(dpt_type.split("-")[2])}
 
     return {}
+
+
+@overload
+def parse_xml_flag(flag: str | None, default: bool) -> bool:
+    ...
+
+
+@overload
+def parse_xml_flag(flag: str | None, default: None = None) -> bool | None:
```

Comment:
Why do we need this?

Reply:
A flag can be set in a 
- ComObjectInstanceRef - user setting
- ComObjectRef - parameter setting
- ComObject - default setting

tag. In latter it is required, the others are optional. We are interested in the first occurrence of that order. So if it is not set we use None. 
Since xml attribute parsing is always `Optional[str]` I used a default of `False` for ComObject - therefore the overload (with a default set the return value is `bool` instead of `bool | None`).

It is used in project_loader.py and application_program_loader.py.